### PR TITLE
Refactoring of construct_building

### PR DESCRIPTION
**Pull request by tibor95**
This is not a functional change, only exclusion some chunks of construct_building into small separate functions.
Right now for discussion,  I will perhaps polish it bit more (comments), but do not plan bigger changes for now.

### CODE REVIEW

**Comment by tibor95** [comment]
the diff is terrible, sorry :(
Maybe I run code formatting over some pieces of code....

**Comment by hessenfarmer** [comment]
Currently i don't have much time so please be patient about my review

**Comment by tibor95** [comment]
do not worry

File: src/ai/defaultai.cc [hunk 1]
```diff
@@ -7388,100 +6967,553 @@ void DefaultAI::print_stats(const Time& gametime) {
...
+			expansion_type.set_expantion_type(ExpansionMode::kBoth);
+		} else {
+			expansion_type.set_expantion_type(ExpansionMode::kSpace);
+		}
+	}
+
+	const bool increase_least_score_limit =
+	   (increase_score_limit_score > management_data.get_military_number_at(45) / 5);
+
+	// static uint16_t concurent_ms_in_constr_no_enemy = 1;
```
Comment (hessenfarmer) [inline src/ai/defaultai.cc]: this could be deleted as it is kept outside in its original place

**Review by hessenfarmer — APPROVED** [review]
Refactoring is complete and looks good.
tested it and found no deviation to master beahviour. I had to merge master locally though.
If CI is green and the few superfluous lines have been deleted It would be good to go.

File: src/ai/defaultai.cc (hunk 1 above)
Comment (tibor95) [reply src/ai/defaultai.cc]: Do you mean the commented out lines? I am deleting them

Comment (hessenfarmer) [reply src/ai/defaultai.cc]: yes exactly

**Comment by hessenfarmer** [comment]
please convert from draft to real as soojn as you are ready to merge this

**Pull request merged**